docs/tutorial.rst

```diff
@@ -869,6 +869,337 @@ of realistic size for real data.
+
+Next,we condition on the mutation not being lost.
+Since in the next step we condition on the mutation being at 80% frequency at
+the end, this is redundant, but it allows the simulation to immediately restart
+from any generation in which the mutation is lost, rather than waiting until the end.
+Note that this conditioning must start one
+generation after the mutation is placed, for which we use
+``stdpopsim.ext.GenerationAfter(T_mut)``.
+We cannot simply specify ``T_mut - 1`` if rescaling is present,
+otherwise the conditioning would start
+at the same generation when the mutation is placed.
+
+.. code-block:: python
+
+    extended_events.append(
+        [
+            stdpopsim.ext.ConditionOnAlleleFrequency(
+                start_time=stdpopsim.ext.GenerationAfter(T_mut),
+                end_time=0,
+                mutation_type_id=mut_id,
+                population_id=0,
+                op=">",
+                allele_frequency=0.0,
+            )
+        ]
+    )
+
+Finally, we condition on the mutation being above 80% at the end of the simulation.
+(The "end" is at time 0, since "time" is in generations before the end of the simulation.)
+
+.. code-block:: python
+
+    extended_events.append(
+        [
+            stdpopsim.ext.ConditionOnAlleleFrequency(
+                start_time=0,
+                end_time=0,
+                mutation_type_id=mut_id,
+                population_id=0,
+                op=">=",
+                allele_frequency=0.8,
+            ),
+        ]
+    )
+
+Now we can simulate, using SLiM of course.
+For comparison, we will run the same simulation
+without selection - i.e., without the "extended events":
+
+.. code-block:: python
+
+    engine = stdpopsim.get_engine("slim")
+    ts_sweep = engine.simulate(
+        model,
+        contig,
+        samples,
+        seed=123,
+        extended_events=extended_events,
+        slim_scaling_factor=100,
+        slim_burn_in=0.1,
+    )
+
+    ts_neutral = engine.simulate(
+        model,
+        contig,
+        samples,
+        seed=123,
+        # no extended events
+        slim_scaling_factor=100,
+        slim_burn_in=0.1,
+    )
+
+Lastly, we can directly compute nucleotide diversity in 10Kb windows for both the
+neutral and sweep simulations and plot them side by side.
+
+.. code-block:: python
+
+    import matplotlib.pyplot as plt
+
+    windows = [w for w in range(0, int(ts_neutral.sequence_length), 10000)]
+    windows.append(int(ts_neutral.sequence_length))
+    neutral_pi = ts_neutral.diversity(windows=windows)
+    sweep_pi = ts_sweep.diversity(windows=windows)
+    plt.plot(neutral_pi, "b", label="neutral")
+    plt.plot(sweep_pi, "r", label="sweep")
+    plt.legend()
+    plt.xlabel("Genomic window")
+    plt.ylabel("Diversity")
+    plt.show()
+
+.. image:: _static/tute-sweep.png
+    :width: 500px
+    :align: center
+    :alt: Plot with nucleotide diversity along the chromosome for simulations with a without a selective sweep.
```

Comment:
Cool! It works!

Reply:
I wanted sphinx to render the plot using matplotlib, but it looks like the page that describes how to do this hasn't been updated in many years and the instructions don't work anymore.

Reply:
if/when we switch to jupyterbook then this does work...